### ISSUE TITLE
psfnight using wrong proccamword

### ISSUE THREAD

**Issue by sbailey** [desihub/desispec#1983]
`desispec.script.submit_night` is using the camword from the first arc as the proccamword for the psfnight, even if that first arc is missing a camera that later arcs have and it could have made a psfnight for that camera.  The result is that we are missing psfnights in some cases where there was enough data to have created one.  Cases from Iron:
```
20210224 - r0 (originally incorrectly listed as b9)
20210510 - b5
20210613 - z3
20211106 - z6
20220101 - z3
20220104 - r2
```

The same bug applies to fiberflatnight too, but is less impactful because we are much less likely to have a single camera failure for flats like we do for arcs.

Instead of using the camword from the first arc, it should include all cameras that have at least 3 good arcs on that night.

**Comment by akremin** [comment]
It's not important, but recording this so that my pull request documentation reflects it: the first night is missing r0 rather than b9. 

Complete list:
```
20210224 - r0
20210510 - b5
20210613 - z3
20211106 - z6
20220101 - z3
20220104 - r2
```

**Comment by sbailey** [comment]
@akremin thanks for catching that, so that we make sure to check the correct camera when comparing to a reprocessing.  I corrected the original posting as well.

FWIW, 20210224 b9 is missing because b9C amp was bad all night, so that is also going to fall back to a default PSF anyway to be used for the 250 fibers on good amps.  But you are correct that r0 is the case that is unnecessarily missing.

**Comment by sbailey** [comment]
@julienguy @akremin I have launched a mini-prod in /global/cfs/cdirs/desi/users/sjbailey/spectro/redux/iron_psfnight covering cals and a subset of tiles for the impacted nights, using Anthony's PR #1984 branch.  Because of the normalization across petals we don't expect bitwise exact match with iron, even for the cameras that previously used a psfnight, but please help compare the differences with iron (psfs, flats, spectra, and final redshifts).

I'll check in the morning about the status of the production and cleanup any failures (it's still running right now but so far so good).

**Comment by julienguy** [comment]
20210224 - r0 :
* wavelength solution for flats is off by 0.2 pixel ~= 0.12A (only issue for flats), PSF shape identical (photometric error < 0.2%)
* no need to reprocess

Method: I am comparing PSF for flat field, for instance,
```
/global/common/software/desi/users/jguy/teststand/bin/plot_psf_comparison_using_specter.py --psf1 /global/cfs/cdirs/desi/users/sjbailey/spectro/redux/iron_psfnight/exposures/20210224/00077869/psf-r0-00077869.fits --psf2 /global/cfs/cdirs/desi/spectro/redux/iron/exposures/20210224/00077869/psf-r0-00077869.fits --fiber 100 --wavelength 7000
```
working on the others ...

**Comment by julienguy** [comment]
20210613 - z3 
 *  wavelength solution for flats is off by 0.1 pixel ~= 0.06A, PSF shape a bit different, but emission line error < 1%
 * no need to reprocess in my opinion

Missing data to continue this test.

**Comment by julienguy** [comment]
20211106 - z6
 * wavelength solution for flats is off by 0.2 pixel ~= 0.12A, psf shape leads to flux bias of 1.4%
 * I would check redshifts to make sure it's ok

20220101 - z3
 * wavelength solution for flats is off by 0.2 pixel ~= 0.12A, psf shape leads to flux bias of 4%
 * need to rerun (!)
 
20220104 - r2
* wavelength solution for flats is off by 0.7 pixel ~ 0.4A , psf shape is identical
* need to check the flat fields ... I verified they are ~OK 
* so we may not need to rerun this one.

20210510 - b5
 * missing /global/cfs/cdirs/desi/users/sjbailey/spectro/redux/iron_psfnight/exposures/20210510/00087888/psf-b5-00087888.fits
 * either re-rerun or abandon this one

**Comment by sbailey** [comment]
The worse case 20220101-z3 has 13% of the ZWARN=0 redshifts changing significantly.  The others are a few percent, with 20210510-b5 being the best at 0.5% differences.  Since we need to rerun several tiles, we'll bite the bullet and rerun all 6 of these after merging + tagging #1984 .

**Comment by akremin** [comment]
The underlying issue was resolved in PR #1984 . Based on the above discussion we removed the six nights of impacted data from iron, including future cumulative redshifts that included that data, and reran all impacted jobs.

This resolves the underlying issue and fixes the impacted processing, so I will close the ticket.
